src/client/app/containers/BarChartContainer.ts

```diff
@@ -78,8 +78,9 @@ function mapStateToProps(state: State) {
 				const hoverText: string[] = [];
 				const readings = _.orderBy(readingsData.readings, ['startTimestamp'], ['asc']);
 				readings.forEach(barReading => {
+					// calculate more precise bar duration
```

Comment:
I would also add this comment above for meters (this is for groups). It might also be good to add why you are subtracting 1 day (because the day ends at midnight of the next day so getting an extra day in the labels).